src/java.base/share/classes/java/lang/Long.java

```diff
@@ -446,36 +447,77 @@ private static void formatUnsignedLong0UTF16(long val, int shift, byte[] buf, in
     }
 
     static String fastUUID(long lsb, long msb) {
+        char[] hex256 = LongCache.HEX256;
+
+        char i0 = hex256[((int) (msb >> 56)) & 0xff];
+        char i1 = hex256[((int) (msb >> 48)) & 0xff];
+        char i2 = hex256[((int) (msb >> 40)) & 0xff];
+        char i3 = hex256[((int) (msb >> 32)) & 0xff];
+        char i4 = hex256[(((int) msb) >> 24) & 0xff];
+        char i5 = hex256[(((int) msb) >> 16) & 0xff];
+        char i6 = hex256[(((int) msb) >> 8) & 0xff];
+        char i7 = hex256[((int) msb) & 0xff];
+        char i8 = hex256[(((int) (lsb >> 56))) & 0xff];
+        char i9 = hex256[(((int) (lsb >> 48))) & 0xff];
+        char i10 = hex256[(((int) (lsb >> 40))) & 0xff];
+        char i11 = hex256[((int) (lsb >> 32)) & 0xff];
+        char i12 = hex256[(((int) lsb) >> 24) & 0xff];
+        char i13 = hex256[(((int) lsb) >> 16) & 0xff];
+        char i14 = hex256[(((int) lsb) >> 8) & 0xff];
+        char i15 = hex256[((int) lsb) & 0xff];
+
+        final byte coder;
+        final int charSize;
+        final int off;
         if (COMPACT_STRINGS) {
-            byte[] buf = new byte[36];
-            formatUnsignedLong0(lsb,        4, buf, 24, 12);
-            formatUnsignedLong0(lsb >>> 48, 4, buf, 19, 4);
-            formatUnsignedLong0(msb,        4, buf, 14, 4);
-            formatUnsignedLong0(msb >>> 16, 4, buf, 9,  4);
-            formatUnsignedLong0(msb >>> 32, 4, buf, 0,  8);
-
-            buf[23] = '-';
-            buf[18] = '-';
-            buf[13] = '-';
-            buf[8]  = '-';
-
-            return new String(buf, LATIN1);
+            coder = LATIN1;
+            charSize = 1;
+            off = 0;
         } else {
-            byte[] buf = new byte[72];
-
-            formatUnsignedLong0UTF16(lsb,        4, buf, 24, 12);
-            formatUnsignedLong0UTF16(lsb >>> 48, 4, buf, 19, 4);
-            formatUnsignedLong0UTF16(msb,        4, buf, 14, 4);
-            formatUnsignedLong0UTF16(msb >>> 16, 4, buf, 9,  4);
-            formatUnsignedLong0UTF16(msb >>> 32, 4, buf, 0,  8);
-
-            StringUTF16.putChar(buf, 23, '-');
-            StringUTF16.putChar(buf, 18, '-');
-            StringUTF16.putChar(buf, 13, '-');
-            StringUTF16.putChar(buf,  8, '-');
-
-            return new String(buf, UTF16);
+            coder = UTF16;
+            charSize = 2;
+            off = ByteOrder.nativeOrder() == ByteOrder.BIG_ENDIAN ? 1 : 0;
         }
+        final byte[] buf = new byte[36 * charSize];
+
+        buf[off] = (byte) (i >> 8);
+        buf[1 * charSize + off] = (byte) i;
```

Comment:
```suggestion
        buf[off] = (byte) (i0 >> 8);
        buf[1 * charSize + off] = (byte) i0;
```
Doesn't compile on my end